### DISCUSSION TITLE
Using supabase without node.js

### DISCUSSION THREAD

**Discussion by stretch07** [supabase/supabase#5473] [Questions] [answered]
Hello everyone! I am building a static site with supabase. It is not built with any js framework, nor with a terminal. I am deploying with Github + Vercel. Does anybody know if it is possible to use supabase without node.js? Or if it isn't possible, would anyone know if it is still possible with Github + Vercel and without a terminal necessary?
Thanks,
Pablo

**Comment by GaryAustin1** [comment, marked as answer]
Not familiar with Vercel, but for a plain js site you can get supabase from cdn:

https://supabase.com/docs/reference/javascript/installing

I've used: <script src="https://cdn.jsdelivr.net/npm/@supabase/supabase-js"></script>

**Comment by stretch07** [comment]
Hello there,
Thank you for your timely response, @GaryAustin1  Loading supabase via CDN is not my primary concern. My main issue here is mainly that I am unsure if exposing my project key is a safe thing. Is it? The init docs say its a public anon key, so I think it's fine. Is it?

**Reply by GaryAustin1** [reply]
No worries on your public anon key.   You protect with RLS on your tables.  No matter how you might hide the public key in node, when the client app makes calls the public key will be emitted in the network request headers so is visible on the client.

Do not use your service_role key though on a static app that should only be used on a server.  You really should not need it for an app though.

I'm working on a svelteKit static app and don't use the service_role key at all in my project.

**Reply by stretch07** [reply]
Awesome, that answered my question. Thanks!

**Reply by stretch07** [reply]
@GaryAustin1 unrelated question: my app is at https://windows99dev.vercel.app/experiments/login, and for some reason it says that createClient is not an export. Do you know why this is?

**Reply by GaryAustin1** [reply]
Have to see your js call, but if using cdn call is like:
var supabase = supabase.createClient(SUPABASE_URL, SUPABASE_KEY)
If using node (and in many supabase examples) is like:
 const supabase = createClient(....)
